cloudimg/aws.py

```diff
@@ -40,18 +57,26 @@ class AWSPublishingMetadata(PublishingMetadata):
                                            interfaces. By default this option
                                            is enabled.
         billing_products (list, optional): Billing product identifiers
+
+        boot_mode (str, optional): The boot mode for booting up the AMI on EC2.
     """
 
     def __init__(self, *args, **kwargs):
         self.ena_support = kwargs.pop('ena_support', True)
         self.sriov_net_support = kwargs.pop('sriov_net_support', 'simple')
         self.billing_products = kwargs.pop('billing_products', None)
+        bmode_str = kwargs.pop('boot_mode', None) or "not_set"
+        self.boot_mode = AWSBootMode[bmode_str]
 
         super(AWSPublishingMetadata, self).__init__(*args, **kwargs)
 
-        # Convert uefi_support bool into AWS specific values
-        self.boot_mode = "uefi-preferred" if self.uefi_support \
-            else "legacy-bios"
+        # Set the UEFI mode when the boolean is used instead of Enum.
+        if (
+            self.boot_mode == AWSBootMode.not_set and
+            self.uefi_support is not None
+        ):
+            self.boot_mode = AWSBootMode.uefi if kwargs.get("uefi_support") \
```

Comment:
there was `"uefi-preferred"` in case self.uefi_support == True`
and that is AWSBootMode.hybrid :-)

Reply:
ouch x.x

thanks! Fixing

Reply:
Now it's pointing to `hybrid` 😄